### PR TITLE
Abstract validation of the options

### PR DESCRIPTION
**Pull request by atimmer**
This way we can add more validation in other locations.

### CODE REVIEW

File: admin/class-admin.php [hunk 1]
```diff
@@ -135,6 +130,36 @@ public function save_settings( $data ) {
...
+	 * @return true|WP_Error true or an error object.
+	 */
+	public function validate_settings() {
+
+		if ( ! empty( $this->options['manual_ua_code_field'] ) ) {
+			$this->options['manual_ua_code_field'] = trim( $this->options['manual_ua_code_field'] );
+			// en dash to minus, prevents issue with code copied from web with "fancy" dash
+			$this->options['manual_ua_code_field'] = str_replace( '–', '-', $this->options['manual_ua_code_field'] );
+
+			if ( ! preg_match( '|^UA-\d{4,}-\d+$|', $this->options['manual_ua_code_field'] ) ) {
```
Comment (CarolineGeven) [inline admin/class-admin.php]: Please add with inline documentation what this regex does

File: admin/class-admin.php [hunk 2]
```diff
@@ -97,21 +97,16 @@ public function save_settings( $data ) {
...
-			$this->options['manual_ua_code_field'] = str_replace( '–', '-', $this->options['manual_ua_code_field'] );
 
-			if ( ! preg_match( '|^UA-\d{4,}-\d+$|', $this->options['manual_ua_code_field'] ) ) {
-
-				$this->add_notification( 'ga_notifications', array(
-					'type'        => 'error',
-					'description' => __( 'The UA code needs to follow UA-XXXXXXXX-X format.', 'google-analytics-for-wordpress' ),
-				) );
+		$validation = $this->validate_settings();
+		if ( is_wp_error( $validation ) ) {
```
Comment (CarolineGeven) [inline admin/class-admin.php]: Perhaps move the error notification to a new function so we can (slowly) clean up this huge function.

File: admin/class-admin.php [hunk 3]
```diff
@@ -135,6 +130,36 @@ public function save_settings( $data ) {
 	}
 
 	/**
+	 * Validates the settings in the `options` attribute, returns an WP_Error object on error
+	 *
+	 * @return true|WP_Error true or an error object.
+	 */
+	public function validate_settings() {
```
Comment (CarolineGeven) [inline admin/class-admin.php]: I think this could be private.

**Comment by CarolineGeven** [comment]
CR1 done

**Comment by atimmer** [comment]
Issues fixed, can be reviewed again.

**Pull request merged**